### DISCUSSION TITLE
New color ramp for terrain - looking for coder for implementing into XCSoar

### DISCUSSION THREAD

**Discussion by MikeBravo15** [XCSoar/XCSoar#1355] [Ideas]
Hallo everybody. I have created a new terrain color ramp for the terrain renderer. My goal was to improve the "vibrant" palette as i find it too dark to be visible in direct sunsine. As a basis I used Hi Contrast schema from my old SeeYou desktop installation and I further modified it to resemble current SeeYou mobile ramp with whitish colors for lowlands. I admit, that I fine tuned it to the heights found in the central Europe.

 ![SeeYouColorRamp](https://github.com/XCSoar/XCSoar/assets/42399455/7800f33a-c8cb-49c6-b80a-a9281ae54a57) 

Output looks like this -
![Screenshot_20240222-181211](https://github.com/XCSoar/XCSoar/assets/42399455/40e588e5-9873-4f15-821b-fbc6c54e838b).

Values for TerrainRenderer.cpp are as follows.

{ // HiContrast
    {0, { 255, 247, 239 }},
    {200, { 225, 225, 225 }},
    {300, { 242, 242, 242 }},
    {400, { 204, 223, 192 }},
    {500, { 153, 199, 144 }},
    {700, { 51, 151, 48 }},
    {900, { 255, 255, 0 }},
    {1500, { 192, 128, 0 }},
    {2000, { 139, 22, 0 }},
    {3500, { 255, 255, 255 }},
    {4500, { 175, 202, 242 }},
    {5500, { 175, 202, 242 }},
    {6000, { 175, 202, 242 }},
  },

Problem is I am not a coder and I am not able to implement it correctly into the XCSoar code as aditional color scheme (I tested with replaced values for one of currently used schemes). Would anybody volunteer to implement it? Thank you for any help

**Comment by lordfolken** [comment]
I can create a pull request for it. What would you call this color ramp?

**Reply by MikeBravo15** [reply]
SeeYou calls it High Contrast and I am quite OK with this. If you have any other name use it

**Reply by lordfolken** [reply]
is this your own work, or is this an exact copy?

**Reply by MikeBravo15** [reply]
Not an exact copy, I tuned it a lot

**Reply by lordfolken** [reply]
The point is, did you take any data from seeyou, or did you create a look a like from scratch?
Because a copy would be a copyright violation.

**Reply by MikeBravo15** [reply]
No, I used a SeeYou only as a tool, because there you can edit color ramps easily with any colors you specify from any altitude to any altitude and you see results quickly - it is look a like from a scratch, not even mimicking anything already present there. No data were extracted directly from SeeYou and preset there is my own work.
![MapCrea](https://github.com/XCSoar/XCSoar/assets/42399455/19145553-bc4f-4cb9-8003-bae6a3926c40)

**Comment by lordfolken** [comment]
https://github.com/XCSoar/XCSoar/compare/master...lordfolken:XCSoar:terrainramp

**Reply by lordfolken** [reply]
#1356

**Reply by MikeBravo15** [reply]
Thank you very much for your kind help

**Reply by lordfolken** [reply]
What's your name/e-mail so we can credit you?

**Reply by MikeBravo15** [reply]
mirek.burdych@gmail.com
